recipes/metview/build.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -e # Abort on error.
+
+export PING_SLEEP=30s
+export WORKDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+export BUILD_OUTPUT=$WORKDIR/build.out
+
+touch $BUILD_OUTPUT
+
+dump_output() {
+    echo Tailing the last 500 lines of output:
+    tail -50000 $BUILD_OUTPUT
+}
+error_handler() {
+    echo ERROR: An error was encountered with the build.
+    dump_output
+    exit 1
+}
+
+# If an error occurs, run our error handler to output a tail of the build.
+trap 'error_handler' ERR
+
+# Set up a repeating loop to send some output to Travis.
+bash -c "while true; do echo \$(date) - building ...; sleep $PING_SLEEP; done" &
+PING_LOOP_PID=$!
+
+export PYTHON=
+export LDFLAGS="$LDFLAGS -L$PREFIX/lib -Wl,-rpath,$PREFIX/lib"
+export CFLAGS="$CFLAGS -fPIC -I$PREFIX/include"
+
+SRC_DIR="$(pwd)"
+BUILD_DIR=$SRC_DIR/../build
+mkdir $BUILD_DIR
+cd $BUILD_DIR
+
+export TMPDIR=/tmp/
+
+if [[ $(uname) == Linux ]]; then
+    export PATH=$PATH:$BUILD_PREFIX/x86_64-conda_cos6-linux-gnu/sysroot/usr/
```

Comment:
Why is this needed?

Reply:
Thought I commented here. I'm having trouble getting cmake to find libpthread.so. I tried setting `CMAKE_LIBRARY_PATH= $BUILD_PREFIX/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib` which worked (as in, cmake found pthread) but the travis builds started started failing at compile time with the error 'cannot find any c preprocessor' which is a bit bizarre.

It seems that using $LDFLAGS isn't working either - cmake still can't find pthread. I'll keep chipping away at it until it works.